nixos/modules/system/boot/loader/generic-extlinux-compatible/extlinux-conf-builder.sh

```diff
@@ -52,7 +52,13 @@ declare -A filesCopied
 
 copyToKernelsDir() {
     local src=$(readlink -f "$1")
-    local dst="$target/nixos/$(cleanName $src)"
+    # Don't copy if on same filesystem, just point to the original.
+    if [[ $(stat -f -c %i $src) = $(stat -f -c %i $target/nixos) ]]; then
```

Comment:
`-c %d` or `-c %D` would be more in line with
https://github.com/NixOS/nixpkgs/blob/5823018b1b27b9675a51a84a7fc9cdd44327fa3e/nixos/modules/system/boot/loader/grub/install-grub.pl#L99-L104
but the filesystem id should be equally safe.

Reply:
Yeah, I actually had `%D` in my personal version but when I was cleaning up for PR, decided to go with the filesystem id since it seems to have a lot more bits. I'm not sure what the "best" answer is. To be truly accurate, maybe it should just attempt to make a hard-link and then remove it :)